runtime/src/accounts_db.rs

```diff
@@ -1266,6 +1266,36 @@ impl AccountsDB {
         }
     }
 
+    pub fn remove_unrooted_slot(&self, remove_slot: Slot) {
+        if self.accounts_index.read().unwrap().is_root(remove_slot) {
+            panic!("Trying to remove accounts for rooted slot {}", remove_slot);
+        }
+
+        let pubkey_sets: Vec<HashSet<Pubkey>> = self.scan_account_storage(
+            remove_slot,
+            |stored_account: &StoredAccount, _, accum: &mut HashSet<Pubkey>| {
+                accum.insert(stored_account.meta.pubkey);
+            },
+        );
+
+        // Purge this slot from the accounts index
+        let mut reclaims = vec![];
+        {
+            let pubkeys = pubkey_sets.iter().flatten();
+            let accounts_index = self.accounts_index.read().unwrap();
+
+            for pubkey in pubkeys {
+                accounts_index.clean_unrooted_entries_by_slot(remove_slot, pubkey, &mut reclaims);
+            }
+        }
+
+        self.handle_reclaims(&reclaims);
+
+        // 1) Remove old bank hash from self.bank_hashes
+        // 2) Purge this slot's storage entries from self.storage
+        self.process_dead_slots();
```

Comment:
nits: how about doing assert to assure storages are removed?

Reply:
added!